tiledb/sm/query/dimension_label/dimension_label_data_query.cc

```diff
@@ -127,6 +127,53 @@ bool is_sorted_buffer_impl(
   return true;
 }
 
+/**
+ * Checks if the input buffer is sorted for variable length buffer.
+ *
+ * TODO: This is a quick-and-dirty implementation while we decide where
+ * sorting is handled for ordered dimension labels. If we keep this design,
+ * we should consider optimizing (parallelizing?) the loops in this check.
+ *
+ * @param buffer Buffer to check for sort.
+ * @param buffer_size Total size of the buffer.
+ * @param increasing If ``true`` check if the data is stricly increasing. If
+ *     ``false``, check if the data is strictly decreasing.
+ */
+bool is_sorted_buffer_str_impl(
+    stats::Stats* stats,
+    const uint64_t* offsets_buffer,
+    const uint64_t* offsets_buffer_size,
+    const char* buffer,
+    const uint64_t* buffer_size,
+    bool increasing) {
+  auto timer_se = stats->start_timer("check_data_sort");
+  uint64_t num_values = *offsets_buffer_size / sizeof(uint64_t);
+  if (increasing) {
```

Comment:
It would be easy to combine those loops using std::greater_equal or std::lesser_equal as a template argument.